cryptography/src/lib.rs

```diff
@@ -147,33 +158,35 @@ pub trait Hasher<const N: usize>: Clone + Send + Sync + 'static {
 
     /// Hash all recorded data and reset the hasher
     /// to the initial state.
-    fn finalize(&mut self) -> Digest<N>;
+    fn finalize(&mut self) -> Self::Digest;
 
     /// Reset the hasher without generating a hash.
     ///
     /// This function does not need to be called after `finalize`.
     fn reset(&mut self);
 
     /// Validate the digest.
-    fn validate(digest: &Digest<N>) -> bool;
+    fn validate(digest: &Self::Digest) -> bool;
 
-    /// Size of the digest in bytes.
-    fn len() -> usize;
+    fn from(data: &[u8]) -> Self::Digest;
```

Comment:
I think there is a way around this (defining on `Digest` directly). Just not sure the best way to do it...